src/server/job_recov_db.c

```diff
@@ -376,6 +378,52 @@ job_save_db(job *pjob, int updatetype)
 	if (pjob->ji_newjob == 1 && updatetype != SAVEJOB_NEW)
 		return (0);
 
+	if (updatetype == SAVEJOB_NEW && pjob->ji_newjob == 0) {
+		/* If new_job flag is not set but still try to save as a new job, changed to SAVEJOB_FULL*/
+		updatetype = SAVEJOB_FULL;
+		sprintf(log_buffer,"job has already been saved earlier as a new job, thus changing to SAVEJOB_FULL");
+		log_joberr(-1, "job_save_db", log_buffer,  pjob->ji_qs.ji_jobid);
+
+#ifndef WIN32
```

Comment:
I would recommend moving this to a new utility function. So that it can be handy for future use.